guides-markdown/events.md

```diff
@@ -0,0 +1,115 @@
+| `address` |  [[WalletKey](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletkey.js)] | [Wallet](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/wallet.js): `_add()`,  |  _wallet_&#8594;[WalletDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletdb.js) (returns&nbsp;parent&nbsp;Wallet,&nbsp;[WalletKey]) |
+
+## Socket Events
+
+Websocket connections in bcoin are handled by two servers, one for `Node` and one for `Wallet`.
+Those servers each have child objects such as `Chain`, `Mempool`, `Pool`, and `WalletDB`, and
+relay events from them out the socket. To receive an event, the socket client must watch a
+channel (such as `chain`, `mempool`, or `auth`) or join the a wallet
+(which would be user-defined like `primary`, `hot-wallet`, or `multisig1`). All wallets can be
+joined at once by joining `'*'`. See more about joining and watching with hooks and calls below.
+
+### Wallet
+
+All wallet events are triggered by a `WalletDB` object, which have may be triggered by its parent
+`TXDB` or `Wallet`. The socket emits the event along with the wallet ID, and whatever "Returns" are listed above.
+
+| Event |
+|-|
+| `tx` |
+| `confirmed` |
+| `unconfirmed` |
+| `conflict` |
+| `balance` |
+| `address` |
+
+### Node
+
+| Event | Returns | Channel | Origin | Original Event |
+|-|-|-|-|-|
+| `chain connect` | [ChainEntry](https://github.com/bcoin-org/bcoin/blob/master/lib/blockchain/chainentry.js)_.toRaw()_ | `chain` | [Chain](https://github.com/bcoin-org/bcoin/blob/master/lib/blockchain/chain.js) | `connect` |
+| `block connect` | [ChainEntry](https://github.com/bcoin-org/bcoin/blob/master/lib/blockchain/chainentry.js)_.toRaw()_, [Block](https://github.com/bcoin-org/bcoin/blob/master/lib/primitives/block.js)_.txs_ | `chain` | [Chain](https://github.com/bcoin-org/bcoin/blob/master/lib/blockchain/chain.js) | `connect` |
+| `chain disconnect` | [ChainEntry](https://github.com/bcoin-org/bcoin/blob/master/lib/blockchain/chainentry.js)_.toRaw()_ | `chain` | [Chain](https://github.com/bcoin-org/bcoin/blob/master/lib/blockchain/chain.js) | `disconnect` |
+| `block disconnect` | [ChainEntry](https://github.com/bcoin-org/bcoin/blob/master/lib/blockchain/chainentry.js)_.toRaw()_ | `chain` | [Chain](https://github.com/bcoin-org/bcoin/blob/master/lib/blockchain/chain.js) | `disconnect` |
+| `tx ` | [TX](https://github.com/bcoin-org/bcoin/blob/master/lib/primitives/tx.js)_.toRaw()_ | `mempool` | [Pool](https://github.com/bcoin-org/bcoin/blob/master/lib/net/pool.js) | `tx` |
+
+## Server Hooks
+
+Certain events can also be sent back to the server from the client to request new data
```

Comment:
Might be worth mentioning and referencing the native bsock methods here as what is used for this and linking to the bsock repo for reference.

Reply:
examples in next commit!

Reply:
:D